src/ml_tooling/baseclass.py

```diff
@@ -277,20 +291,22 @@ def test_estimators(
             )
             challenger_estimator = cls(estimator)
             result = challenger_estimator.score_estimator(
-                data=data, metric=metric, cv=cv
+                data=data, metrics=metrics, cv=cv
             )
             results.append(result)
-            if log_dir:
-                result.log_estimator(log_dir)
 
-        results.sort(reverse=True)
-        best_estimator = results[0].estimator
+        results = ResultGroup(results).sort()
+
+        if log_dir:
+            results.log_estimator(pathlib.Path(log_dir))
+
+        best_estimator = results[0].model
```

Comment:
Should we retrain the best estimator? Or at least give the option?

Reply:
I can add that as a keyword flag

Reply:
Done